File: app/src/main/java/com/tangem/data/network/BlockchairApi.java
package com.tangem.data.network;

import com.tangem.data.network.model.BlockchairAddressResponse;
import com.tangem.data.network.model.BlockchairSendBody;
import com.tangem.data.network.model.BlockchairStatsResponse;
import com.tangem.data.network.model.BlockchairTransactionResponse;

import io.reactivex.Completable;
import io.reactivex.Single;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface BlockchairApi {
    @GET(Server.ApiBlockchair.Method.ADDRESS)
    Single<BlockchairAddressResponse> getAddress(@Path("blockchain") String blockchain, @Path("address") String address);

    @GET(Server.ApiBlockchair.Method.TRANSACTION)
    Single<BlockchairTransactionResponse> getTransaction(@Path("blockchain") String blockchain, @Path("transaction") String transaction);

    @GET(Server.ApiBlockchair.Method.STATS)
    Single<BlockchairStatsResponse> getStats(@Path("blockchain") String blockchain);

    @POST(Server.ApiBlockchair.Method.PUSH)
    Completable sendTransaction(@Path("blockchain") String blockchain, @Body BlockchairSendBody body);
}

File: blockchain/src/main/java/com/tangem/blockchain/network/blockchair/BlockchairApi.kt
package com.tangem.blockchain.network.blockchair

import com.squareup.moshi.JsonClass
import retrofit2.http.*

interface BlockchairApi {
    @GET("{blockchain}/dashboards/address/{address}")
    suspend fun getAddressData(
            @Path("address") address: String,
            @Path("blockchain") blockchain: String,
            @Query("key") key: String
    ): BlockchairAddress

    @GET("{blockchain}/dashboards/transaction/{transaction}")
    suspend fun getTransaction(
            @Path("transaction") transactionHash: String,
            @Path("blockchain") blockchain: String,
            @Query("key") key: String
    ): BlockchairTransaction

    @GET("{blockchain}/stats")
    suspend fun getBlockchainStats(
            @Path("blockchain") blockchain: String,
            @Query("key") key: String
    ): BlockchairStats

    @POST("{blockchain}/push/transaction")
    suspend fun sendTransaction(
            @Body sendBody: BlockchairBody,
            @Path("blockchain") blockchain: String,
            @Query("key") key: String
    )
}

@JsonClass(generateAdapter = true)
data class BlockchairBody(val data: String)